src/catalua.cpp

```diff
@@ -736,12 +736,18 @@ void update_globals( lua_State *L )
 {
     LuaReference<player>::push( L, g->u );
     luah_setglobal( L, "player", -1 );
+    // luah_setglobal pushes an extra copy of the global data before storing it,
+    // but here the original value isn't needed once the global data has been
+    // saved.
+    lua_pop( L, 1 );
```

Comment:
That only works if luah_setglobal is called with index -1, otherwise the wrong item will be popped off the stack. luah_setglobal can be used to store values not at the top of the stack (by passing an index other than -1), and as-is it does not alter the stack--the caller is responsible for any needed cleanup. I don't know why luah_setglobal was written to do that, but I didn't want to assume that its current behavior was undesired. Since all current calls only use an index of -1, I could eliminate luah_setglobal entirely.

Reply:
Makes sense. Thanks.